### PR TITLE
Changes PhaseAfterStep to take the name of the previous phase

### PR DESCRIPTION
**Pull request by colings86**
This changes the way the phase after step is built so its key has the
phase name of the phase that preceeds it rather than the phase that
follows it. This is more intuitive to the user since the index is in
the warm phase until the after condition for the cold phase is met.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java [hunk 1]
```diff
@@ -203,7 +207,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
 
         if (phase != null) {
-            Step.StepKey afterStepKey = new Step.StepKey(phase.getName(), "pre-" + lastStepKey.getAction(), "after");
+            // The very first after step is in a phase before the hot phase so call this "new"
+            Step.StepKey afterStepKey = new Step.StepKey("new", PhaseAfterStep.NAME, PhaseAfterStep.NAME);
```
Comment (talevy) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java]: So, for documentation purposes, our phases are now

1. new
2. hot *(optional)*
3. warm *(optional)*
4. cold *(optional)*
5. delete *(optional)*
6. completed

Comment (colings86) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java]: yes, although we should be clear in the docs that `new` and `completed` are not real phases in the fact that you cannot add actions to them

**Review by talevy — CHANGES_REQUESTED** [review]
there are some assertions made in rest tests expecting the pre-pre-readonly step

https://github.com/talevy/elasticsearch/blob/ca9f307b0a72286aaa6595f2a0087c1ddd6f75c6/x-pack/plugin/src/test/resources/rest-api-spec/test/index_lifecycle/20_move_to_step.yml#L133-L135

this  needs to be updated for tests to pass

**Review by talevy — APPROVED** [review]
merging latest index-lifecycle should fix CI on this PR

**Pull request merged**